packages/server-wallet/src/objectives/objective-manager.ts

```diff
@@ -54,87 +66,58 @@ export class ObjectiveManager {
 
     while (attemptAnotherProtocolStep) {
       await this.store.lockApp(channelToLock, async tx => {
-        let executeProtocol: () => ChannelState.ProtocolResult;
-        let protocolState: OpenChannelProtocol.ProtocolState | CloseChannelProtocol.ProtocolState;
+        let nextAction: ChannelState.ProtocolResult;
+        let protocolState: SupportedProtocolState;
 
         if (objective.type === 'OpenChannel') {
-          protocolState = await OpenChannelProtocol.getOpenChannelProtocolState(
+          protocolState = await getOpenChannelProtocolState(
             this.store,
             objective.data.targetChannelId,
             tx
           );
-          executeProtocol = () =>
-            OpenChannelProtocol.protocol(protocolState as OpenChannelProtocol.ProtocolState);
+          nextAction = recordFunctionMetrics(
+            openChannelProtocol(protocolState as OpenChannelProtocolState),
+            this.timingMetrics
+          );
         } else if (objective.type === 'CloseChannel') {
-          protocolState = await CloseChannelProtocol.getCloseChannelProtocolState(
+          protocolState = await getCloseChannelProtocolState(
             this.store,
             objective.data.targetChannelId,
             tx
           );
-          executeProtocol = () =>
-            CloseChannelProtocol.protocol(protocolState as CloseChannelProtocol.ProtocolState);
+          nextAction = recordFunctionMetrics(
```

Comment:
Having `recordFunctionMetrics` sprinkled everywhere seems pretty bad for readability. This PR increases the number of these wrappers. Is there a strategy we can employ to improve this? Hide them inside the imported functions implementations?

Reply:
It's not clear to me whether we even still need these. In this case it's wrapping a pure function, which as far as I can tell doesn't do hashing or signing, so I'm not sure what the point it. Will ask the others when they're up.